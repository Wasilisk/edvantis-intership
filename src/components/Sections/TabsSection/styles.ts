import styled from "styled-components";
import {TabsStyles} from "../../Tabs/styles";
import {ProductCardStyles} from "../../Cards/ProductCard/styles";

export const TabsSectionStyles = styled.div`
  max-width: 1200px;
  margin: 0 auto;
  padding: 40px 24px 50px 24px;
  
  .text-info {
    padding: 0 15px;
    
    .title {
      font-size: 38px;
      color: #333333;
      line-height: 1.2;
      font-weight: 600;
      letter-spacing: 0;
      margin: 0;
    }

    .description {
      font-size: 18px;
      line-height: 1.56;
      letter-spacing: 0;
      color: #333333;
      margin: 25px 0 0 0;
    }
  }

  .tab-panel {
    width: 100%;
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    grid-row-gap: 30px;

    ${ProductCardStyles} {
      padding: 0 15px;
    }
  }
  
  ${TabsStyles} {
    margin-top: 40px;
  }

  @media (max-width: 1024px) {
    .tab-panel {
      grid-template-columns: repeat(2, 1fr);
    }
  }
  
  @media (max-width: 768px) {
    .text-info {
      padding: 0;
    }
  }

  @media (max-width: 425px) {
    .tab-panel {
      grid-template-columns: repeat(1, 1fr);
    }
  }
`;


