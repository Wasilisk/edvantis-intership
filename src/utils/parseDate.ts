const month = ["January","February","March","April","May","June","July","August","September","October","November","December"];

export const parseDate = (timestamp: number) => {
    const date = new Date(timestamp);

    return `${month[date.getMonth()]} ${('0' + date.getDate()).slice(-2)}, ${date.getFullYear()}`;
}