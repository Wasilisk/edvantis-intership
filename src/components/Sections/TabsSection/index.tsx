import React, {useState} from 'react';
import {TabsSectionStyles} from "./styles";
import {ProductType, TabsSectionType} from "../../../models";
import Tabs from "../../Tabs";
import Tab from "../../Tabs/Tab";
import TabPanel from "../../Tabs/TabPanel";
import ProductCard from "../../Cards/ProductCard";

const TabsSection = ({title, description, tabs}: TabsSectionType) => {
    const [value, setValue] = useState<number | string>(tabs[0].id)

    return (
        <TabsSectionStyles id="#favourites">
            <div className="text-info">
                <h4 className="title">{title}</h4>
                <p className="description">{description}</p>
            </div>
            <Tabs>
                {
                    tabs.map(tab => {
                        const clickHandler = () => setValue(tab.id);

                        return <Tab
                            key={tab.id}
                            label={tab.label}
                            isActive={value === tab.id}
                            onClick={clickHandler}
                        />
                    })
                }
            </Tabs>
            {
                tabs.map(tab => <TabPanel key={tab.id} value={value} index={tab.id}>
                    {
                        Array.isArray(tab.items) && tab.items.map((item: ProductType) => <ProductCard
                                {...item}
                                key={item.id}
                                buttonVariant="outlined"
                            />
                        )
                    }
                </TabPanel>)
            }
        </TabsSectionStyles>
    );
};

export default TabsSection;