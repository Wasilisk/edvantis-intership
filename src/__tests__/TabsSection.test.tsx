import {render} from "./utils/store-wrapper";
import TabsSection from "../components/Sections/TabsSection";
import {mockTabsSection} from "./__mock__/mock-tabs-section";
import {fireEvent} from "@testing-library/react";

describe('TabsSection component', () => {
    it('should render component', () => {
        const {container, getByText, getByTestId, asFragment} = render(<TabsSection {...mockTabsSection}/>);

        const title = getByText(mockTabsSection.title);
        expect(title).toBeDefined();

        const description = getByText(mockTabsSection.description);
        expect(description).toBeDefined();

        const tabs = getByTestId('tabs');
        expect(tabs).toBeDefined();
        expect(tabs.childElementCount).toBe(3);

        expect(container.firstChild!.childNodes.length).toBe(5);

        expect(asFragment()).toMatchSnapshot();
    })
    it('should change content when tab clicked', () => {
        const {getByTestId} = render(<TabsSection {...mockTabsSection}/>);

        const firstTabPanel = getByTestId(`tab-panel-${mockTabsSection.tabs[0].id}`)
        expect(firstTabPanel).toBeDefined();

        const tabs = getByTestId('tabs');
        fireEvent.click(tabs.childNodes[1]);

        const secondTabPanel = getByTestId(`tab-panel-${mockTabsSection.tabs[1].id}`)
        expect(secondTabPanel).toBeDefined();
    })
})