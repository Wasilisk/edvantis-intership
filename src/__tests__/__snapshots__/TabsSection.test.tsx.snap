// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsSection component should render component 1`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf ePRWOu"
  >
    <div
      class="text-info"
    >
      <h4
        class="title"
      >
        Title
      </h4>
      <p
        class="description"
      >
        Description
      </p>
    </div>
    <div
      class="sc-bczRLJ gnmHVj"
    >
      <div
        class="tabs"
        data-testid="tabs"
      >
        <div
          class="sc-hKMtZM bIqTAQ"
          value="1"
        >
          <span>
            Tab 1
          </span>
        </div>
        <div
          class="sc-hKMtZM fMPRoN"
          value="2"
        >
          <span>
            Tab 2
          </span>
        </div>
        <div
          class="sc-hKMtZM fMPRoN"
          value="3"
        >
          <span>
            Tab 3
          </span>
        </div>
      </div>
    </div>
    <div
      class="tab-panel"
      role="tabpanel"
    >
      <div
        class="sc-gsnTZi isgJGv"
      >
        <a
          class="card-image"
          href="/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-eCYdqJ hnhvuX"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-gsnTZi isgJGv"
      >
        <a
          class="card-image"
          href="/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-eCYdqJ hnhvuX"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-gsnTZi isgJGv"
      >
        <a
          class="card-image"
          href="/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-eCYdqJ hnhvuX"
        >
          Order Now
        </button>
      </div>
      <div
        class="sc-gsnTZi isgJGv"
      >
        <a
          class="card-image"
          href="/lord-armchair-miedel-home"
        >
          <img
            alt="Lord armchair, Miedel Home"
            src="https://e-c.storage.googleapis.com/res/5f731c835a5a2667cb27290e/480"
          />
        </a>
        <div
          class="card-text"
        >
          <a
            href="/lord-armchair-miedel-home"
          >
            Lord armchair, Miedel Home
          </a>
          <p
            class="highlight"
          >
            120.50USD
          </p>
        </div>
        <button
          class="sc-eCYdqJ hnhvuX"
        >
          Order Now
        </button>
      </div>
    </div>
    <div
      class="tab-panel"
      hidden=""
      role="tabpanel"
    />
    <div
      class="tab-panel"
      hidden=""
      role="tabpanel"
    />
  </div>
</DocumentFragment>
`;
